### ISSUE TITLE
How to close lightbox overlay through javascript

### ISSUE THREAD

**Issue by jfbourgon** [wet-boew/wet-boew#6933]
I'm able to open modal inline content throught javascript using the following code
```javascript
$document.trigger( "open.wb-lbx", [
			[
				{
					src: "#inline_content_modal",
					type: "inline"
				}
			],
			true
		]);
```
I'm wondering how I could also invoke closing of the opened overlay throught javascript.

**Comment by pjackson28** [comment]
You can use <code>$.magnificPopup.close();</code>. According to the Magnific Popup documentation it will close the currently open Lightbox.

**Comment by jfbourgon** [comment]
I already tried this code in the first place but I was getting a javascript error about $.magnificPopup being undefined.

I then tried to import the library manually with something like:
```html
<script src="./theme-gcwu-fegc/unmin/js/deps/jquery.magnific-popup.js"></script>
```
I got ride of the error but the `$.magnificPopup.close();` call had no effect on the dialog box. 

Looks like I can't get an handle on `$.magnificPopup` from outside the WET plugin.

**Comment by LaurentGoderre** [comment]
You shouldn't include the file manually. Is it possible that you your `$.magnificPopup.close();` was not actually executed when a lightbox was open and before the plugin was initialized?

**Comment by jfbourgon** [comment]
Are you suggesting that the `$.magnificPopup.close();` directive is called too soon after the open call so this is why it is yet undefined.

I can try to but some delay between each call. How can I ensure the lightbox dialog is completed when this one is trigger by javascript?

**Comment by LaurentGoderre** [comment]
So you want to close a lightbox without any user action?

**Comment by jfbourgon** [comment]
Yes. To give a bit of context my web application is doing ajax call that may take few second to execute. I was expecting to open a modal lightbox with some kind of loading animation prior to execute the ajax call and then close it on success response . Maybe there could be other alternatives?

**Comment by pjackson28** [comment]
If you want to close a modal then that is covered in the documentation with <code>popup-modal-dismiss</code>:
* Identifies the button that is to be used to close a lightbox behaving like a modal dialog.
* Add popup-modal-dismiss to the link or button that will be used to close the lightbox behaving like a modal dialog.

That is the recommended way of dealing with modal closure. If that isn't sufficient then you should wait on the <code>wb-ready.wb-lbx</code> event before trying to close anything (to ensure the Magnific Popup object is present). It will be fired after the Lightbox has finished initializing.

Here is some JS that should help:
```js
$( document ).on( "wb-ready.wb-lbx", function( event ) {
});
```

**Comment by jfbourgon** [comment]
Here is the testing code I put up so far. This should cause the dialog to open on WET initialization and then being instantly closed once ready (this is not exactly the behaviour I'm looking for but this would be a good starting point). In my case, the `wb-ready.wb-lbx` event is never invoked.
```javascript
$( document ).on( "wb-ready.wb", function( event ) {   
    $( document ).trigger( "open.wb-lbx", [
        [
            {
                src: "#loading-pane",
                type: "inline"
            }
        ],
        true
    ] );
});

$( document ).on( "wb-ready.wb-lbx", function( event ) {
    $.magnificPopup.close();
});
```

**Comment by pjackson28** [comment]
Try moving the wb-ready.wb-lbx handler before the wb-ready.wb handler. JS in the same file is technically synchronous so likely all the open.wb-lbx is finishing before the wb-ready.wb-lbx handler ever gets established.

**Comment by pjackson28** [comment]
To be even safer (in the case of another Lightbox on the same page), try putting wb-ready.wb-lbx inside the wb-ready.wb handler right before the open.wb-lbx call.

**Comment by jfbourgon** [comment]
Based on your suggestion I'm now using this code but without luck. The `wb-ready.wb-lbx` event is still never invoked and the dialog remains opened.

```javascript
$( document ).on( "wb-ready.wb", function( event ) {   
    $( document ).on( "wb-ready.wb-lbx", function( event ) {
        $.magnificPopup.close();
    });   
    
    $( document ).trigger( "open.wb-lbx", [
        [
            {
                src: "#loading-pane",
                type: "inline"
            }
        ],
        true
    ] );
} );
```

**Comment by LaurentGoderre** [comment]
I don't think that's right. Event handlers should not be nested.

**Comment by pjackson28** [comment]
They can be nested, particularly when you don't want to be listening for one event (wb-ready.wb-lbx) until another has already fired (wb-ready.wb). If you don't nest it, then the close will fire repeatedly on any page that has wb-lbx elements which are initializing before the wb-ready.wb. Keep in mind that this is an edge case that should not be implemented like this in practice (just a proof of concept).

As for why it is not working, seems that the wb-ready is not being fired when opening a Lightbox through JS. PR #6934 fixes that.

**Comment by pjackson28** [comment]
Note that I tested your code with the fix from PR #6934 and it worked perfectly.

**Comment by jfbourgon** [comment]
Thanks for your great support! Really appreciated.
